File: web/src/app/types/ImportJob.ts
export type ImportJobStatus = 'Pending' | 'Processing' | 'Completed' | 'Aborted';
export type ImportJobItemStatus = 'Pending' | 'Processing' | 'Succeeded' | 'Failed' | 'Skipped';

export interface ImportJobItem {
  id: string;
  sightTitle: string;
  sightDescription?: string;
  status: ImportJobItemStatus;
  errorMessage?: string;
  sightId?: string;
  categoryName?: string;
  tags?: string;
  latitude?: number;
  longitude?: number;
  image350Url?: string;
}

export interface ImportJob {
  id: string;
  status: ImportJobStatus;
  totalCount: number;
  processedCount: number;
  succeededCount: number;
  failedCount: number;
  skippedCount: number;
  abortRequested: boolean;
  createdAt: string;
  completedAt?: string;
  items: ImportJobItem[];
}

export interface StartImportResponse {
  jobId: string;
  totalCount: number;
  skippedCount: number;
  skipped: string[];
}

File: Api/Models/ImportJob.cs
namespace Api.Models;



public class ImportJob
{
    public Guid Id { get; set; }
    public ImportJobStatus Status { get; set; }
    public int TotalCount { get; set; }
    public int ProcessedCount { get; set; }
    public int SucceededCount { get; set; }
    public int FailedCount { get; set; }
    public int SkippedCount { get; set; }
    public bool AbortRequested { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public List<ImportJobItem> Items { get; set; } = [];
}

public enum ImportJobStatus { Pending, Processing, Completed, Aborted }
